lib/chef/provider/aws_subnet.rb

```diff
@@ -6,22 +6,40 @@ class Chef::Provider::AwsSubnet < Chef::Provider::AwsProvider
   action :create do
     fail "Can't create a Subnet without a CIDR block" if new_resource.cidr_block.nil?
 
-    if existing_subnet == nil
-      converge_by "Creating new Subnet #{fqn} in VPC #{new_resource.vpc} in #{new_resource.region_name}" do
+    if existing_subnet
+      subnet = existing_subnet
+      # TODO update things
```

Comment:
Can you actually set the map_public_ip_on_launch attribute on launch?  I didn't think that was possible: http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/EC2/SubnetCollection.html#create-instance_method

I'm not morally opposed to setting things on create, but I think the logic for update is identical to create in this case (the thing must already exist before you can set its attribute).